### ISSUE TITLE
Local provider fails to copy on Windows from non-elevated console

### ISSUE THREAD

**Issue by mvromer** [hashicorp/terraform#25583]
### Terraform Version

```
Terraform v0.13.0-beta3
```

### Terraform Configuration Files

```terraform
terraform {
  required_providers {
    example = {
      source = "example.local/mvromer/example"
    }
  }
}

provider example {}
```

### Debug Output

https://gist.github.com/mvromer/d5fecf749a4727efdd174ea31839f2d5

### Crash Output

No crash.

### Expected Behavior

When running from a non-elevated console on Windows, Terraform should have correctly installed the local `example` provider declared in the Terraform configuration above.

### Actual Behavior

Terraform fails to initialize because it could not install the `example` provider. It reports an error saying it failed to symlink or copy the local provider to the destination directory underneath the `.terraform` folder because it could not find the destination path.

When running `terraform init` from an *elevated* console, initialization succeeds because Terraform is able to symlink the local provider to the destination under the `.terraform` directory.

### Steps to Reproduce

I've actually put together a minimal reproducible test case that exhibits this, including a much more detailed bug report in my [mvromer/terraform-windows-local-provider-copy-bug](https://github.com/mvromer/terraform-windows-local-provider-copy-bug) GitHub repo. That repo's README has detailed reproduction steps, but the synopsis is given here:

1. Clone the above repo
2. Run the `build.bat` script to build the local provider and install it under the configuration's root module under `terraform.d\plugins\...`.
3. From a **non-elevated** console, change into the `configuration` directory and run `terraform init`. Initialization should fail.
4. Now from an **elevated** console, run `terraform init` again. Initialization should succeed and successfully install the local provider.

### Additional Context

I actually did a fair amount of spelunking through the Terraform code to try to debug this myself. It was how I finally figured out Terraform installs local providers differently on Windows if `terraform init` is run from an elevated console versus a non-elevated one.

The above repo's README actually goes into a lot more detail, but the tl;dr is that I think [installFromLocalDir](https://github.com/hashicorp/terraform/blob/v0.13.0-beta3/internal/providercache/package_install.go#L117) needs to be updated so that if symlink creation fails, the function should create the target directory before attempting to do a [deep copy via the call to copydir.CopyDir](https://github.com/hashicorp/terraform/blob/v0.13.0-beta3/internal/providercache/package_install.go#L172). The function currently will [only create the parent of the target directory](https://github.com/hashicorp/terraform/blob/v0.13.0-beta3/internal/providercache/package_install.go#L159) (which is correct for the symlink case), but `copydir.CopyDir` [requires both the source and destination directories to exist](https://github.com/hashicorp/terraform/blob/v0.13.0-beta3/internal/copydir/copy_dir.go#L13).

As mentioned in the above repo, I *think* this can be fixed by modifying the `installFromLocalDir` function to essentially add an `os.Mkdir(absNew, 0755)` call immediately before the call to `copydir.CopyDir`. This should create the target directory only if symlink creation fails and satisfy the pre-condition for `copydir.CopyDir`. However, I don't know if there are other things that such a change might impact for the worse.

### References

None.

**Comment by mildwonkey** [comment]
Hi @mvromer ! Thank you for the detailed issue report - I labeled this issue 'explained' but want to emphasize that you are the person who explained it, and found the code paths and likely solution. I haven't reproduced this myself but your explanation makes sense and is exactly what I'd try. 
Since you're already in this code base, please feel free to open a pull request with this change, otherwise i'll make sure this is properly labelled so we (or someone) can get it fixed.
Cheers!

**Comment by mvromer** [comment]
Hi @mildwonkey, I was about to start work on a PR for this based on your feedback, but it looks like this issue was fixed between 0.13 RC1 and 0.13 GA in this [commit](https://github.com/hashicorp/terraform/commit/440543f42712d3ecb53d2504475e581b0c4ac090). I just tested this locally against my test repo, and it seems the original bug is now fixed. Thanks though for the feedback. I'll go ahead and close this since it looks to be resolved.
